### PR TITLE
Implementation of issue 5

### PR DESCRIPTION
**Pull request by pfif**
Implementation of #5 

My final goal (as stated in the issue's description) was to add a command to pass automatically account files from a period to the other.

That necessitated a few steps : 
* 61979ad6a0e41de0a3e82dbf32b400db62d4d926 I realize that the parser file had two job, parsing account files and parsing the command line. So I split those two functions in two files, io.py for the first job and entrypoint.py for the second one
* c256e662f11adbb5cd8ee323be71abef8e1c9e1d I added subcommands to my program, in a git-styled format
* 45a149e43c443c16a87f279f14bf2993961d41b0 I had to make the Account class instatiable without a knowing current amount of money on the account. And... I had to refactor all the tests for them to test this functionality (ugh... tough one)
* 17bd833e1b483ea5c97f0edfc0e854da5eea69e2 Finally, I added two feature : One that format the account without changing any value, another that format the account and that pass the account to the next period. I wish I had time to figure out how AST work to create the account with them. Instead of that I just used the __repr__ functions of the Operations. It is not the best solution, but it works.

### CODE REVIEW

File: pyrestriction/model.py [hunk 1]
```diff
@@ -123,10 +132,22 @@ def amount(self):
...
+
+        classname = self.__class__.__name__
+
+        return "{classname}({arguments})".format(classname=classname, arguments=arguments)
+
 class SavingOperation(Operation):
     """Save money during one period"""
     def __init__(self, amount):
         super(SavingOperation, self).__init__(amount)
+        self._amount = amount
```
Comment (johnzeringue) [inline pyrestriction/model.py]: This smells a bit. Why do you have the same private variable in the superclass and the subclass?

**Comment by johnzeringue** [comment]
It's kind of hard for me to look through the diff because your lines are so long. `endpoint.py` isn't particularly well named, because it's really a command line interface, right? A few good integration tests would help me understand what you're actually looking to do here. Maybe look into [Cucumber](https://cucumber.io).

File: pyrestriction/model.py (hunk 1 above)
Comment (pfif) [inline pyrestriction/model.py]: Yeah, what the hell was I thinking ...

**Comment by pfif** [comment]
Well, I need this now, so I am going to pass it. Again, I am sorry forgot about this. Thanks for your review anyway ! :+1:

**Pull request closed without merging**